app.js

```diff
@@ -2,44 +2,33 @@ angular.module('PathOfDamage', [])
 .controller('Damage', function ($scope, DataService) {
   $scope.sections = DataService.getSections();
 
-  $scope.hits = [{hit: 100}, {hit: 500}, {hit: 1000}, {hit: 2000}, {hit: 3000},
-    {hit: 4000}, {hit: 5000}, {hit: 7500}, {hit: 10000}];
+  $scope.hits = [{hit: 100}, {hit: 500}, {hit: 1000}, {hit: 2000}, {hit: 3000}, {hit: 5000}, {hit: 7500}, {hit: 10000}];
   $scope.resistance = 75;
-  $scope.healthPool = 7000;
+  $scope.healthPool = 5000;
 
-  $scope.updateHit = function (index, skipStringify) {
-    if ($scope.hits[index].hit === null) {
-      $scope.hits.splice(index, 1);
-    }
-    if (index > 0 && $scope.hits[$scope.hits.length - 1].hit !== null) {
-      $scope.hits.push({hit: null});
-    }
-    if ($scope.hits[index]) {
-      $scope.hits[index] = calcDamage($scope.hits[index].hit);
-    }
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
-    }
+  $scope.clear = function () {
+    window.location.search = '';
+    window.location.reload();
   };
 
-  $scope.add = function (table, skipStringify) {
+  $scope.add = function (table) {
     var lastEntry = table.values[table.values.length - 1];
-    if (!lastEntry || lastEntry.name !== '' || lastEntry.value !== null) {
+    if (lastEntry.name !== '' || lastEntry.value !== null) {
       table.values.push({
         enabled: true,
         name: '',
         value: null
       });
     }
-    $scope.updateTotal(table, skipStringify);
+    $scope.updateTotal(table);
   };
 
-  $scope.delete = function (table, index, skipStringify) {
+  $scope.delete = function (table, index) {
     table.values.splice(index, 1);
-    $scope.updateTotal(table, skipStringify);
+    $scope.updateTotal(table);
   };
 
-  $scope.updateTotal = function (table, skipStringify) {
+  $scope.updateTotal = function (table, skipUpdates) {
     if (table.total !== undefined) {
```

Comment:
No! This one specifically needs to check for undefined. Tables with no rows have 0 totals, but some tables that don't HAVE totals (like more and less tables, because they are multiplicative not additive), and those are specifically the ones that shouldn't calculate.